app/models/course/assessment/question/programming.rb

```diff
@@ -162,10 +162,6 @@ def polyglot_language_version
     language.name.split[1]
   end
 
-  def language_valid_for_codaveri?
-    codaveri_language_whitelist.include?(language.type.constantize)
-  end
-
```

Comment:
Removing this breaks admin page `coursemology2/app/views/course/admin/codaveri_settings/edit.json.jbuilder`

You could possibly make an alias

```rb
  def language_valid_for_codaveri?
    CodaveriAsyncApiService.language_valid_for_codaveri?(language)
  end
```

Then line 267 can revert to `if !language_valid_for_codaveri?`

Reply:
This will be fixed by https://github.com/Coursemology/coursemology2/pull/7532